### PR TITLE
Bach: Raise exception when window partitioning has no order by columns.

### PR DESCRIPTION
**Pull request by KathiaBarahona**
As mentioned in https://github.com/objectiv/objectiv-analytics/pull/1064, BigQuery raises exceptions when some analytics functions have no order by clause (which makes sense). Now, an exception is raised for all `Series.window_*` and `DataFrame.rolling` and `DataFrame.expanding`, since we should avoid generating any non-deterministic result or any problem when executing the final query.

### CODE REVIEW

File: bach/tests/functional/bach/test_df_window.py [hunk 1]
```diff
@@ -186,6 +187,10 @@ def test_windowing_windows(engine):
     for w in [p0,p1,p2,p3]:
         bt.inhabitants.window_row_number(window=w).to_pandas()
 
+    with pytest.raises(ValueError, match=r'Window must me sorted when applying'):
```
[the remark this answers is not in the document]
Comment (jansentom) [reply bach/tests/functional/bach/test_df_window.py:190]: this could be a unit test?

File: bach/bach/dataframe.py [hunk 2]
```diff
@@ -1824,13 +1838,21 @@ def rolling(self, window: int,
...
             end_boundary = WindowFrameBoundary.FOLLOWING
 
-        index = list(self._group_by.index.values()) if self._group_by else []
-        group_by = Window(group_by_columns=index,
-                          order_by=self._order_by,
-                          mode=mode,
-                          start_boundary=start_boundary, start_value=start_value,
-                          end_boundary=end_boundary, end_value=end_value,
-                          min_values=min_periods)
+        window_params = {
```
Comment (jansentom) [inline bach/bach/dataframe.py:1841]: why are you leaving out the order_by?

File: bach/bach/dataframe.py [hunk 3]
```diff
@@ -1860,14 +1894,21 @@ def expanding(self,
...
         end_value = None
 
-        index = list(self._group_by.index.values()) if self._group_by else []
-        group_by = Window(group_by_columns=index,
-                          order_by=self._order_by,
-                          mode=mode,
-                          start_boundary=start_boundary, start_value=start_value,
-                          end_boundary=end_boundary, end_value=end_value,
-                          min_values=min_periods)
+        window_params = {
```
Comment (jansentom) [inline bach/bach/dataframe.py:1897]: why are you leaving out the order_by here?

File: bach/tests/functional/bach/test_df_window.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (KathiaBarahona) [reply bach/tests/functional/bach/test_df_window.py:190]: oops typo when raising the error, good catch!

File: bach/tests/functional/bach/test_df_window.py [hunk 4]
```diff
@@ -170,6 +170,7 @@ def frame_clause_equals(expected, **kwargs):
 def test_windowing_windows(engine):
     ## Just test that different windows don't generate SQL errors. Logic will be checked in different tests.
     bt = get_df_with_test_data(engine, full_data_set=True)
+    bt = bt.sort_index()
 
     # no sorting, no partition
```
[the remark this answers is not in the document]
Comment (KathiaBarahona) [reply bach/tests/functional/bach/test_df_window.py:175]: actually need to update the comments for all those cases.

File: bach/bach/dataframe.py (hunk 2 above)
Comment (KathiaBarahona) [reply bach/bach/dataframe.py:1841]: Will raise an error, because `DataFrame.window` method already sets it

File: bach/bach/dataframe.py (hunk 3 above)
Comment (KathiaBarahona) [reply bach/bach/dataframe.py:1897]: `DataFrame.window` already sets it :P

**Pull request merged**